### PR TITLE
Field Variable setValue() looks up variable.

### PR DESCRIPTION
**Pull request by marisaleung**

### CODE REVIEW

**Comment by RoboErikG** [comment]
As discussed offline, try to lookup by id first, then by name and if found in either case set the value to the id and the text to the name. If not found, the current default of setting both to the value passed in is fine.

**Comment by marisaleung** [comment]
Change made!

File: tests/jsunit/field_variable_test.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+function test_fieldVariable_setValueNoVariable() {
+  // Expect the fieldVariable value to be set to the passed in string. No error
+  // should be thrown.
+  fieldVariable_setUp();
+  var fieldVariable = new Blockly.FieldVariable('name1');
+  var mockBlock = {'workspace': workspace,
+    'isShadow': function(){return false;}};
+  fieldVariable.setSourceBlock(mockBlock);
+  fieldVariable.setValue('id1');
+  assertEquals('id1', fieldVariable.getText());
```
Comment (RoboErikG) [inline tests/jsunit/field_variable_test.js]: also check that the value was set to 'id1'

Comment (marisaleung) [inline tests/jsunit/field_variable_test.js]: Done thank you!

Comment (marisaleung) [inline tests/jsunit/field_variable_test.js]: Done

**Pull request merged**